www/js/splash/remoteNotifyHandler.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * This module deals with handling specific push messages that open web pages
+ * or popups. It does not interface with the push plugin directly. Instead, it
+ * assumes that another module (currently `pushnotify`) deals with the plugin
+ * interface and emits a CLOUD_NOTIFICATION_EVENT when a push notification is
+ * received.
+ *
+ * This allows us to decouple the push handling logic from push notification
+ * interface. Note that the local notification is not currently decoupled since
+ * it only supports redirection to a specific app page. If the local
+ * notification handling gets more complex, we should consider decoupling it as well.
+ */
+import { EVENT_NAMES, subscribe } from '../customEventHandler';
+import { addStatEvent, statKeys } from '../plugin/clientStats';
+import { displayErrorMsg, logDebug } from '../plugin/logger';
+
+const options = 'location=yes,clearcache=no,toolbar=yes,hideurlbar=yes';
+
+/*
+TODO: Potentially unify with the survey URL loading
+*/
+/**
+ * @function launches a webpage
+ * @param url to open in the browser
+ */
+const launchWebpage = function (url) {
+  // THIS LINE FOR inAppBrowser
+  let iab = window['cordova'].InAppBrowser.open(url, '_blank', options);
+};
+
+/*
+TODO: replace popup with something with better UI
+*/
+
+/**
+ * @function launches popup
+ * @param title string text for popup title
+ * @param text string text for popup bode
+ */
+const launchPopup = function (title, text) {
+  // THIS LINE FOR inAppBrowser
+  displayErrorMsg(text, title);
+};
+
+/**
+ * @callback for cloud notification event
+ * @param event that triggered this call
+ */
+const onCloudNotifEvent = (event) => {
+  const data = event.detail;
+  addStatEvent(statKeys.NOTIFICATION_OPEN).then(() => {
+    console.log('Added ' + statKeys.NOTIFICATION_OPEN + ' event. Data = ' + JSON.stringify(data));
+  });
+  logDebug('data = ' + JSON.stringify(data));
+  if (
+    data.additionalData &&
+    data.additionalData.payload &&
+    data.additionalData.payload.alert_type
+  ) {
+    if (data.additionalData.payload.alert_type == 'website') {
```

Comment:
@shankari Are we ever going to re-implement external web launches via push notification? If not, we can simplify

Reply:
I think we should re-implement all the different types of push notification. I have pitched the external web launches, in particular, as mechanism for providing incentives for deployments with autogenerated tokens. The program admin can send a link to an amazon gift card directly to the participant though a push notification, without having to know their email/phone number etc.
Follow on to https://github.com/e-mission/e-mission-docs/issues/976